resources/Microsoft.VSCode.DSC/Microsoft.VSCode.DSC.psm1

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version Latest
+
+enum Ensure
+{
+    Absent
+    Present
+}
+
+#region DSCResources
+[DSCResource()]
+class VSCodeExtension
+{
+    [DscProperty(Key)]
+    [string] $Name
+
+    [DscProperty()]
+    [string] $Version
+
+    [DscProperty()]
+    [Ensure] $Ensure = [Ensure]::Present
+
+    [DscProperty(NotConfigurable)]
+    [hashtable] $InstalledExtensions = @{}
+
+    [VSCodeExtension] Get()
+    {
+        Assert-VSCode
+
+        $currentState = [VSCodeExtension]::new()
+        $currentState.Ensure = [Ensure]::Absent
+        $currentState.Name = $this.Name
+        $currentState.Version = $this.Version
+        $extensionList = (Invoke-VSCode -Command "--list-extensions --show-versions") -Split [Environment]::NewLine
+
+        foreach ($extension in $extensionList)
+        {
+            $info = $extension -Split '@'
+            $extensionName = $info[0]
+            $extensionVersion = $info[1]
+            $currentState.InstalledExtensions[$extensionName] = $extensionVersion
+        }
+
+        foreach ($extension in $currentState.InstalledExtensions.Keys)
+        {
+            if ($extension -eq $this.Name)
+            {
+                if ($null -ne $this.Version)
+                {
+                    if ($this.Version -eq $currentState.InstalledExtensions[$this.Name])
+                    {
+                        $currentState.Ensure = [Ensure]::Present
+                    }
+                }
+                else
+                {
+                    $currentState.Ensure = [Ensure]::Present
+                }
+
+                break
+            }
+        }
+        
+        return $currentState
+    }
+
+    [bool] Test()
+    {
+        $currentState = $this.Get()
+        return $currentState.Ensure -eq $this.Ensure
+    }
+
+    [void] Set()
+    {
+        if ($this.Test())
+        {
+            return
+        }
+
+        if ($this.Ensure -eq [Ensure]::Present)
+        {
+            $extensionArg = $this.Name
+
+            if ($null -ne $this.Version)
+            {
+                $extensionArg += "@$($this.Version)"
+            }
+
+            Invoke-VSCode -Command "--install-extension $($extensionArg)"
+        }
+        else
+        {
+            Invoke-VSCode -Command "--uninstall-extension $($this.Name)"
+        }
+
+    }
+}
+
+function Assert-VSCode
+{
+    # Refresh session $PATH value before invoking 'code.exe'.
+    $env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine") + ";" + [System.Environment]::GetEnvironmentVariable("Path","User")
```

Comment:
This seems like it could cause issues.  What if I updated `%PATH%` before invoking this DSC and now you've lost my change that pointed you to code.exe?

Reply:
I did this initially because the path was not getting updated if the previous configuration unit was install VSCode. I changed it so that it points directly to the expected code.cmd path which is this `$env:LocalAppData\Programs\Microsoft VS Code\bin\code.cmd` to avoid messing with the path values.